### PR TITLE
[SRBT-314] Implement Invoicing

### PR DESCRIPTION
**Pull request by Cutaiar**
# [SRBT-314] Implement Invoicing

**Main Changes**

- Implement invoice dashboard
- Implement invoice creation form
- Implement public invoice view (for clients to pay)
- Adds an invoice link to the sidebar

**Storybook changes**
- Adds a `sorbet` background option globally to storybook to render components against the same background we use in the app
- Add global mock providers decorator and update existing stories to use it
- Mock next router globally
- Add [msw](https://mswjs.io/) to add the ability to mock network requests in storybook

**Sidebar changes**
- Add a feature flag for wallet balance in sidebar
- Moves sidebar open state out of redux (using local state in `<Header/>`) and removes unused redux state
- Add a story for sidebar

**Additional Changes**
- Adds a sorbet button variant
- Adds a lighter skeleton variant
- Fix `<Header/>` logo size, loading with priority, and better alt text
- Generalizes `useGenerateQRCode` for use in invoices. Make the share profile QR code the size it is in design
- Sets up and uses [sonner](https://ui.shadcn.com/docs/components/sonner) for toasts
- Animate `<FormMessage/>` by default (with variant escape hatch)

**Notes**

- I'm colocating invoicing specific components with their routes in the app directory
- I'm colocating invoicing specific hooks there too
- I'm choosing to make some slight differences from design to use tailwind color tokens correctly

# Screenshots

### CODE REVIEW

File: src/app/invoices/components/create/invoice-form-context.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+const InvoiceFormContext = createContext<InvoiceFormContextType | null>(null);
+InvoiceFormContext.displayName = 'InvoiceFormContext';
+
+export const InvoiceFormProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  // Client Details
+  const clientDetailsQueryState = {
```
Comment (ChristopherPark3) [inline src/app/invoices/components/create/invoice-form-context.tsx:46]: Ahh so this is what you meant when you said managing state in url looks just like regular react state?

File: src/app/invoices/components/dashboard/invoice-sheet.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        setOpen(open);
+      }}
+    >
+      <SheetContent className='flex h-full w-full flex-col justify-between gap-12 p-6'>
+        <SheetHeader>
+          <SheetTitle className='text-sm font-medium'>
+            {invoice.toName}
+          </SheetTitle>
+          <SheetDescription>
+            <VisuallyHidden>Description goes here</VisuallyHidden>
```
Comment (ChristopherPark3) [inline src/app/invoices/components/dashboard/invoice-sheet.tsx:81]: Did you mean to leave this in?

Comment (ChristopherPark3) [reply src/app/invoices/components/dashboard/invoice-sheet.tsx:81]: The 'Description goes here' part

**Review by ChristopherPark3 — APPROVED** [review]
Awesome stuff Dillon! excited to see this go in

File: src/app/invoices/components/dashboard/invoice-sheet.tsx (hunk 2 above)
Comment (Cutaiar) [reply src/app/invoices/components/dashboard/invoice-sheet.tsx:81]: Ah thanks for caching that. Will fix.

File: src/app/invoices/components/create/invoice-form-context.tsx (hunk 1 above)
Comment (Cutaiar) [reply src/app/invoices/components/create/invoice-form-context.tsx:46]: Yep! Great lib https://nuqs.47ng.com/

File: src/api/invoices/invoices.ts
Comment (audreyostrom) [inline src/api/invoices/invoices.ts:1]: This is a learning moment more so for myself to make sure I'm adhering to the general style guidelines, but do you prefer functions in the api folder be labelled with params and return values in addition to a simple description?

File: src/app/invoices/components/dashboard/utils.tsx
Comment (audreyostrom) [inline src/app/invoices/components/dashboard/utils.tsx:1]: Could the console.log be removed here

File: src/app/invoices/hooks/use-create-invoice.ts
Comment (audreyostrom) [inline src/app/invoices/hooks/use-create-invoice.ts:1]: I think the console.log could be removed here if there's a toast notification right after displaying the error

**Review by audreyostrom — APPROVED** [review]
Looks good and clean to me! I only noticed super minor things that don't impact anything.

File: src/app/invoices/components/dashboard/utils.tsx
Comment (Cutaiar) [reply src/app/invoices/components/dashboard/utils.tsx:1]: I considered it but I'd like to revisit this in the future. Also, we have the build set up to remove console logs in production :)

File: src/api/invoices/invoices.ts
Comment (Cutaiar) [reply src/api/invoices/invoices.ts:1]: Either way, generally, I find them to be a bit redundant to the blurb above, but trying it out here to see if I like it.

**Pull request merged**